ui/src/app/pages/workspace/workspace-library.tsx

```diff
@@ -67,55 +64,78 @@ const LibraryTab: React.FunctionComponent<{
     </Clickable>;
   };
 
+class ProfileState {
+  profile: Profile;
+  reload: Function;
+}
+
+class CurrentTab {
+  title: string;
+  icon: string;
+}
+
+class Props {
+  profileState: ProfileState;
+  enablePublishedWorkspaces: boolean;
+}
+
+class State {
+  currentTab: CurrentTab;
+  errorText: string;
+  workspaceList: WorkspacePermissions[];
+  publishedWorkspaces: WorkspacePermissions[];
+  featuredWorkspaces: WorkspacePermissions[];
+  workspacesLoading: boolean;
+}
+
 export const WorkspaceLibrary = withUserProfile()
-(class extends React.Component<
-  {profileState: { profile: Profile, reload: Function } },
-  {currentTab: {title: string, icon: string}, errorText: string,
-    workspaceList: WorkspacePermissions[], featuredWorkspaces: WorkspacePermissions[],
-    workspacesLoading: boolean}> {
+(class extends React.Component<Props, State> {
   constructor(props) {
     super(props);
     this.state = {
       currentTab: libraryTabEnums.FEATURED_WORKSPACES,
       errorText: '',
       featuredWorkspaces: [],
+      publishedWorkspaces: [],
       workspaceList: [],
       workspacesLoading: true
     };
   }
 
+  libraryTabs = (this.props.enablePublishedWorkspaces || environment.enablePublishedWorkspaces)
+    ? [
+      libraryTabEnums.FEATURED_WORKSPACES,
+      libraryTabEnums.PUBLISHED_WORKSPACES
+    ]
+    : [
+      libraryTabEnums.FEATURED_WORKSPACES
+    ];
+
   async componentDidMount() {
-    this.loadPublishedWorkspaces();
-    this.loadFeaturedWorkspaces();
+    this.updateListedWorkspaces();
   }
 
   componentDidUpdate(prevProps, prevState) {
     // Reload libraries when switching tabs
     if (this.state.currentTab !== prevState.currentTab) {
-      this.updateListedWorkspaces(this.state.currentTab);
+      this.updateListedWorkspaces();
     }
   }
 
-  updateListedWorkspaces(tabEnum) {
-    switch (tabEnum) {
-      case libraryTabEnums.PUBLISHED_WORKSPACES:
-        this.loadPublishedWorkspaces();
-        break;
-      default:
-        this.loadFeaturedWorkspaces();
-        break;
-    }
+  async updateListedWorkspaces() {
```

Comment:
Naming nit: While reading through the code I was getting pretty confused between this method and the "loadWorkspaceList" method. It would be super nice to have more easily distinguishable names. Some ideas for renaming this method:

- UpdateAllWorkspaces
- UpdateWorkspaces
- LoadWorkspacesAndConfig